### ISSUE TITLE
Many external dependencies

### ISSUE THREAD

**Issue by tredoe** [owenthereal/gotask#14]
I think that gotask is getting very heavy with so many external dependencies.

Please, keep it light because I find a penalty performance at running a command from gotask vs "go test".

**Comment by tredoe** [comment]
In addition, with all those external dependencies gotask could not be added to the Go's toolchain.

**Comment by owenthereal** [comment]
This is a full list of dependencies `gotask` use:

``` plain
github.com/codegangsta/cli
github.com/kballard/go-shellquote
```

There's plan to make `gotask` depend on no one :smile_cat:. And I'm looking to solve it soon.

Regarding the performance penalty at running a command, I'm suspecting you're running in the "on the fly" mode. If you don't [compile the tasks](https://github.com/jingweno/gotask#compiling-tasks) before hand, `gotask` will compile them and run them, say `gotask say-hello`. You could see this happens when feeding it with the `--debug` flag like `gotask say-hello --debug`. This behaviour is similar to `go test`. The only difference is `go test` uses an [internal build API](http://golang.org/src/cmd/go/build.go?#L359) and `gotask` uses the `go build` command. I'm wondering though if the performance issue comes from that.

Do you happen to know anyone from the Go team I could talk to about this? I'd really love to use that internal API if possible.

**Comment by tredoe** [comment]
Yes, the performance penalty is because I use the mode "on fly".

Well, you could copy that internal build API to your project. Besides of question about it in the mailing list -- I think that it is the best way to contact with one of the guys of the Go team

**Comment by tredoe** [comment]
In fact, the code could be extracted to create a package that can be used by other projects. But before of that, I thinkt that it would be good to know the opinion about the Go team respect to if they would do public such API.

**Comment by tredoe** [comment]
https://groups.google.com/forum/#!topic/golang-nuts/0FBoVbLyfkw

**Comment by owenthereal** [comment]
:+1: thanks for posting it!

**Comment by owenthereal** [comment]
I posted to G+ as well just in case people are not checking G Group often: https://plus.google.com/116384151198560836084/posts/Z79vRdLY9dd

**Comment by owenthereal** [comment]
@kless in case you're not aware, the [issue](https://code.google.com/p/go/issues/detail?id=6920&thanks=6920&ts=1386615062) has been accepted by the Go team :smile_cat:. I'm gonna leave this issue open until I fix the dependency issue

**Comment by tredoe** [comment]
That's great, thanks!

**Comment by tredoe** [comment]
@jingweno, after of update a driver (binding to C SQL engine --github.com/mattn/go-sqlite3--), I've checked that it has got greater speed and being used via gotask, so the issue was not in your tool.

**Comment by owenthereal** [comment]
@kless That's what I would assume since `gotask` only use `go build` so the speed will be as fast as `go build`. But I still want to minimize the dependencies so that `gotask` can be more portable.
